### DISCUSSION TITLE
why does useSearchParams need to be wrapped in Suspense to avoid client-side-deopt, but not usePathname?

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#58577] [App Router] [answered]
when using `useSearchParams` in a client component, i get a warning message that the "Entire page deopted into client-side rendering.", linking to this explanation in the docs: https://nextjs.org/docs/messages/deopted-into-client-rendering which suggests to wrap the component in a Suspense boundary.

however, with a client component which uses `usePathname` (but not `useSearchParams`), i get no warning.

both hooks are "Client Component hooks", so it is not 100% clear to me why one triggers the warning and the other does not?

**Comment by icyJoseph** [comment, marked as answer]
> If a route is [statically rendered](https://nextjs.org/docs/app/building-your-application/rendering/server-components#static-rendering-default), calling useSearchParams() will cause the tree up to the closest [Suspense boundary](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#example) to be client-side rendered.
> 
> This allows a part of the page to be statically rendered while the dynamic part that uses searchParams is client-side rendered.

It has to do with static rendering, if the page is dynamic then doesn't happen right? And that's because during build time, there's no queries, in the URL param, these ought to be provided by the client when browsing. Pathnames in the other hand, have to be provided explicitly during build time.

**Reply by devsmaxcode** [reply]
I need to access search parameters only after hydration. However, wrapping useSearchParams in Suspense is not an option in my case, as it's used at the root level and causes layout shifts and other issues, making it impossible to replicate a full-page experience.

Is there a way to retrieve search parameters without using Suspense? Currently, I'm relying on a workaround, but I'd prefer a cleaner solution.
